src/Functions/reinterpretAs.cpp

```diff
@@ -535,29 +535,625 @@ using FunctionReinterpretAsFixedString = FunctionReinterpretAs<DataTypeFixedStri
 
 REGISTER_FUNCTION(ReinterpretAs)
 {
-    factory.registerFunction<FunctionReinterpretAsUInt8>();
-    factory.registerFunction<FunctionReinterpretAsUInt16>();
-    factory.registerFunction<FunctionReinterpretAsUInt32>();
-    factory.registerFunction<FunctionReinterpretAsUInt64>();
-    factory.registerFunction<FunctionReinterpretAsUInt128>();
-    factory.registerFunction<FunctionReinterpretAsUInt256>();
-    factory.registerFunction<FunctionReinterpretAsInt8>();
-    factory.registerFunction<FunctionReinterpretAsInt16>();
-    factory.registerFunction<FunctionReinterpretAsInt32>();
-    factory.registerFunction<FunctionReinterpretAsInt64>();
-    factory.registerFunction<FunctionReinterpretAsInt128>();
-    factory.registerFunction<FunctionReinterpretAsInt256>();
-    factory.registerFunction<FunctionReinterpretAsFloat32>();
-    factory.registerFunction<FunctionReinterpretAsFloat64>();
-    factory.registerFunction<FunctionReinterpretAsDate>();
-    factory.registerFunction<FunctionReinterpretAsDateTime>();
-    factory.registerFunction<FunctionReinterpretAsUUID>();
-
-    factory.registerFunction<FunctionReinterpretAsString>();
-
-    factory.registerFunction<FunctionReinterpretAsFixedString>();
-
-    factory.registerFunction<FunctionReinterpret>();
+    FunctionDocumentation::Description description_reinterpretAsUInt8 = R"(
+Performs byte reinterpretation by treating the input value as a value of type UInt8.
+Unlike [`CAST`](#cast), the function does not attempt to preserve the original value - if the target type is not able to represent the input type, the output is meaningless.
+)";
+    FunctionDocumentation::Syntax syntax_reinterpretAsUInt8 = "reinterpretAsUInt8(x)";
+    FunctionDocumentation::Arguments arguments_reinterpretAsUInt8 = {
+        {"x", "Value to byte reinterpret as UInt8.", {"(U)Int*", "Float*", "Date", "DateTime", "UUID", "String", "FixedString"}}
+    };
+    FunctionDocumentation::ReturnedValue returned_value_reinterpretAsUInt8 = {"Returns the reinterpreted value `x`.", {"UInt8"}};
+    FunctionDocumentation::Examples examples_reinterpretAsUInt8 = {
+    {
+        "Usage example",
+        R"(
+SELECT
+    toInt8(257) AS x,
```

Comment:
Strange example. `toInt8(257)` produces an internal overflow.

Can we do this (basically the same example as in docs of function CAST):

```cpp
SELECT
    toInt8(-1) AS val,
    toTypeName(val),
    reinterpretAsUInt8(val) AS res,
    toTypeName(res);
```

?

There will still be an overflow but during re-interpretation as UInt8 ... which is what the description above warns about.